### ISSUE TITLE
[BUG] Application Gateway for Containers ALB Controller helm chart contains a namespace resource in the template

### ISSUE THREAD

**Issue by zioproto** [Azure/AKS#3857]
**Describe the bug**

The Application Gateway for Containers ALB Controller helm chart contains a namespace resource in the template.

The Namespace resource should never be part of the Chart template. The namespace should be handled with the `--namespace` and `--create-namespace` helm command line options.

**To Reproduce**

You can see that the `--namespace test` option is passed but a `azure-alb-system` namespace is created anyways.

```
helm template test oci://mcr.microsoft.com/application-lb/charts/alb-controller      --version 0.4.023971 --namespace test
Pulled: mcr.microsoft.com/application-lb/charts/alb-controller:0.4.023971
Digest: sha256:499e70ac154f5d6aafd180d0c1d6f69e28af555c18a99c0938eeb7fb845f5601
---
# Source: alb-controller/templates/common.yaml
apiVersion: v1
kind: Namespace
metadata:
  name: azure-alb-system
[..CUT..]
```

**Expected behavior**

**Environment (please complete the following information):**
 -  Helm Version: v3.12.3
 - Kubernetes version v1.26.6

**Additional context**
This is causing problems to me because I cant install this Helm Chart with the Helm Terraform provider:

```
data azurerm_user_assigned_identity azure_alb_identity {
    name                = "azure-alb-identity"
    resource_group_name = "istio-aks"
}

resource "helm_release" "alb-controller" {
  chart            = "alb-controller"
  namespace        = "azure-alb-system"
  create_namespace = "false"
  name             = "alb-controller"
  version          = "0.4.023971"
  repository       = "oci://mcr.microsoft.com/application-lb/charts/"
  atomic           = "true"
  values = [
    yamlencode({
      albController = {
        podIdentity = {
          clientID = data.azurerm_user_assigned_identity.azure_alb_identity.client_id
        }
      }
    })
  ]
}
```

This fails with `Error: create: failed to create: namespaces "azure-alb-system" not found`

If I change to ` create_namespace = "true"` then there is a conflict between the namespace defined in the Chart and the one that helm tries to create:

```
│ Error: release alb-controller failed, and has been uninstalled due to atomic being set: 1 error occurred:
│       * namespaces "azure-alb-system" already exists
```

**Solution**

Drop the following lines from the chart in `common.yaml` file:

```
apiVersion: v1
kind: Namespace
metadata:
  name: {{ include "albController.namespace" . }}
```
Drop the following lines in `values.yaml`
```
  # -- Namespace to deploy alb-controller components in.
  namespace: "azure-alb-system"
```

**Extra feedback**
I was able to download the source code of the helm chart with:
```
helm pull oci://mcr.microsoft.com/application-lb/charts/alb-controller      --version 0.4.023971
```
But I was not able to identify a GitHub repo where this code is versioned in public. It would be easier to propose a PR to a public repo to fix this issue.

thank you

**Comment by zioproto** [comment]
I found also another small problem following the instructions here:
https://learn.microsoft.com/en-us/azure/application-gateway/for-containers/quickstart-deploy-application-gateway-for-containers-alb-controller?tabs=install-helm-windows

Installing with the following command:
```
helm install alb-controller oci://mcr.microsoft.com/application-lb/charts/alb-controller \
     --version 0.4.023971 \
     --set albController.podIdentity.clientID=$(az identity show -g $RESOURCE_GROUP -n azure-alb-identity --query clientId -o tsv)
```

The pods go to the default `azure-alb-system` namespace. However the helm metadata about the installation goes to the current active namespace, that for me it is the default namespace:

```
% kubectl get secret -n default
NAME                                   TYPE                 DATA   AGE
sh.helm.release.v1.alb-controller.v1   helm.sh/release.v1   1      2d1h
```

**Comment by kaarthis** [comment]
Gentle ping and bump on this.

**Comment by philwelz** [comment]
> azure-alb-system

you can (for now) remove the namespace and create_namespace from the helm_release resource in Terraform as the helm chart will by default create and install the release to the namespace `azure-alb-system`.

**Comment by JackStromberg** [comment]
Hey folks,

Thank you for bubbling this up.  We released our next version of ALB Controller (0.5.024542), which has the changes needed to address this.  See the following link for more information: https://learn.microsoft.com/azure/application-gateway/for-containers/quickstart-deploy-application-gateway-for-containers-alb-controller?tabs=install-helm-windows#for-new-deployments

Cheers!
Jack

#please-close
